src/resolution/composite-arns-resolver.ts

```diff
@@ -67,6 +74,82 @@ export class CompositeArNSResolver implements NameResolver {
       registryCache,
       networkProcess,
     });
+    this.limit = pLimit(maxConcurrentResolutions);
+    this.resolverTimeoutMs = resolverTimeoutMs;
+  }
+
+  private async resolveWithResolver(
+    resolver: NameResolver,
+    name: string,
+    baseArNSRecordFn: () => Promise<any>,
+    isLastResolver: boolean,
+  ): Promise<NameResolution | undefined> {
+    try {
+      const resolutionPromise = resolver
+        .resolve({
+          name,
+          baseArNSRecordFn,
+        })
+        .then((resolution) => {
+          if (resolution.resolvedAt !== undefined) {
+            this.resolutionCache.set(
+              name,
+              Buffer.from(JSON.stringify(resolution)),
+            );
+            this.log.info('Resolved name', { name, resolution });
```

Comment:
nit: add `resolvedBy: resolver.constructor.name` to logs. 

this could also be a prometheus metric